.github/workflows/release.yml

```diff
@@ -0,0 +1,62 @@
+name: Release
+
+on:
+  release:
+    types: [released]
+  workflow_dispatch:
+    inputs:
+      TAG_NAME:
+        description: "Tag name"
+        required: true
+
+env:
+  REGISTRY: ghcr.io
+  TAG_NAME: ${{ github.event.inputs.TAG_NAME || github.event.release.tag_name }}
+
+jobs:
+  retag-docker-image:
+    runs-on: ubuntu-latest
+    name: Push updated Docker image
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set environment variables
+        run: |
+          echo VERSION=${TAG_NAME#v} >> $GITHUB_ENV
+          echo IMAGE_NAME=$REGISTRY/$(echo ${GITHUB_REPOSITORY,,}) >> $GITHUB_ENV
+          echo COMMITED_AT=$(git show -s --format=%cI `git rev-parse HEAD`) >> $GITHUB_ENV
+          echo REVISION=$(git rev-parse --short HEAD) >> $GITHUB_ENV
+
+      - name: Collect Docker image metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.created=${{ env.COMMITED_AT }}
+            org.opencontainers.image.version=v${{ env.VERSION }}
+            org.opencontainers.image.maintainer=GeoWerkstatt GmbH <support@geowerkstatt.ch>
+          tags: |
+            type=semver,pattern=v{{version}}
+
+      - name: Log in to the GitHub container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner  }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          build-args: |
+            VERSION=${{ env.VERSION }}
+            REVISION=${{ env.REVISION }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=registry,ref=${{ env.IMAGE_NAME }}:edge
+          cache-to: type=inline
```

Comment:
Q: Reicht es nicht aus, wenn wir das Docker-Image mit dem Tag `edge` aus der Registry holen und dann re-taggen / pushen? So können wir uns das Builden des Docker-Images sparen.

Reply:
Mit den Optionen _cache-from/to_ werden bereits sämtliche wiederverwendbaren Layer aus dem letzten _edge_-Build verwendet und nur noch neu getagt. 

```
#16 [build 6/8] RUN dotnet restore "EWS.csproj"
#16 CACHED
#17 [build 8/8] RUN dotnet publish "EWS.csproj"   -c Release   -p:VersionPrefix=0.1.4   -p:SourceRevisionId=a83bd19   -o /app/publish
#17 CACHED
#18 [final 2/4] WORKDIR /app
#18 CACHED
#19 [final 4/4] COPY --from=build /app/publish /app
#19 CACHED

#20 exporting to image
#20 exporting layers done
#20 writing image sha256:6f9348e41c489038931f8f99f3d765b67b1f33c18ce16a7c6597d503fa5ea027 done
#20 naming to ghcr.io/geowerkstatt/ews-boda:v0.1.4 done
#20 naming to ghcr.io/geowerkstatt/ews-boda:latest done
```
Beim Release werden nur noch die Tags zum bereits bestehenden edge Image hinzugefügt. Das Image ändert nicht.
![image](https://user-images.githubusercontent.com/1365121/175550100-34d15ba4-9a97-4722-ac59-a67baeb24057.png)

_Edge_ wird bei jedem Commit auf _main_ neu erstellt. Wenn wir beim Releasen das letzte edge verwenden würden, müssten wir sicherstellen dass in der Zwischenzeit auf main kein weiterer Commit passiert. Es wäre ausserdem auch nicht möglich eine spezifische Version zu releasen.